Test/VimCoreTest/InterpreterTest.cs

```diff
@@ -1317,13 +1317,95 @@ public void RHSCanBeArbitraryExpression()
             }
 
             [WpfFact]
-            public void RHSCanBeBinaryExpression()
+            public void RHSCanBeBinaryAddExpression()
             {
                 Create("");
                 ParseAndRun("let x=1+2");
                 AssertValue("x", 3);
             }
 
+            [WpfFact]
+            public void RHSCanBeBinarySubtractExpression()
+            {
+                Create("");
+                ParseAndRun("let x=1-2");
+                AssertValue("x", -1);
+            }
+
+            [WpfFact]
+            public void RHSCanBeBinaryMultiplyExpression()
+            {
+                Create("");
+                ParseAndRun("let x=4*2");
+                AssertValue("x", 8);
+            }
+
+            [WpfFact]
+            public void RHSCanBeBinaryDivideExpression()
+            {
+                Create("");
+                ParseAndRun("let x=24/3");
+                AssertValue("x", 8);
+            }
+
+            [WpfFact]
+            public void RHSCanBeBinaryModuloExpression()
+            {
+                Create("");
+                ParseAndRun("let x=20%7");
+                AssertValue("x", 6);
+            }
+
+            [WpfFact]
+            public void RHSCanBeBinaryDivideExpressionAndHandleDivByZero()
+            {
+                Create("");
+                _variableMap["x"] = VariableValue.NewNumber(7);
+                ParseAndRun("let x=24/0");
+                AssertValue("x", 7);

```

Comment:
Should validate here that the error is issued for divide by zero.

Reply:
Done.
I've found that regardless of what we issue in RunBinaryExpression - the error is overwritten later, so I modified TestableStatusUtil a bit to test this case. I'll probably need to replace that by moq later, but that would require updating a lot of tests so I wanted to separate it into a different PR.